Example failure where I saw this happen: https://github.com/apache/airflow/actions/runs/12223335598/job/34094868625?pr=44686#step:9:670

```python
________________ TestWatchedSubprocess.test_reading_from_pipes _________________
[gw1] linux -- Python 3.9.21 /usr/local/bin/python
task_sdk/tests/execution_time/test_supervisor.py:113: in test_reading_from_pipes
    assert captured_logs == [
E   AssertionError: assert equals failed
E         'level': 'info',                 'level': 'info',            
E         'logger': 'task',                'logger': 'task',           
E         'timestamp': '2024-11-07T12      'timestamp': '2024-11-07T12 
E     :34:56.078901Z',                 :34:56.078901Z',                
E       },                               },                            
E       {                                {                             
E         'chan': 'stdout',                'chan': 'stderr',           
E         'event': 'Message split acr      'event': 'stderr message',  
E     oss two writes',                                                 
E         'level': 'info',                 'level': 'error',           
E         'logger': 'task',                'logger': 'task',           
E         'timestamp': '2024-11-07T12      'timestamp': '2024-11-07T12 
E     :34:56.078901Z',                 :34:56.078901Z',                
E       },                               },                            
E       {                                {                             
E         'chan': 'stderr',                'chan': 'stdout',           
E         'event': 'stderr message',       'event': 'Message split acr 
E                                      oss two writes',                
E         'level': 'error',                'level': 'info',            
E         'logger': 'task',                'logger': 'task',           
E         'timestamp': '2024-11-07T12      'timestamp': '2024-11-07T12 
E     :34:56.078901Z',                 :34:56.078901Z',                
E       },                               },                            
E       {                                {                             
E         'event': 'An error message'      'event': 'An error message' 
E     ,                                ,

```

--- a/hatch_build.py
+++ b/hatch_build.py
@@ -267,6 +267,7 @@ DEVEL_EXTRAS: dict[str, list[str]] = {
         "pytest-mock>=3.12.0",
         "pytest-rerunfailures>=13.0",
         "pytest-timeouts>=1.2.1",
+        "pytest-unordered>=0.6.1",
         "pytest-xdist>=3.5.0",
         "pytest>=8.2,<9",
         "requests_mock>=1.11.0",

--- a/task_sdk/pyproject.toml
+++ b/task_sdk/pyproject.toml
@@ -80,6 +80,7 @@ dev-dependencies = [
     "kgb>=7.1.1",
     "pytest-asyncio>=0.24.0",
     "pytest-mock>=3.14.0",
+    "pytest-unordered>=0.6.1",
     "pytest>=8.3.3",
 ]
 

--- a/task_sdk/tests/execution_time/test_supervisor.py
+++ b/task_sdk/tests/execution_time/test_supervisor.py
@@ -32,6 +32,7 @@ from unittest.mock import MagicMock
 import httpx
 import psutil
 import pytest
+from pytest_unordered import unordered
 from uuid6 import uuid7
 
 from airflow.sdk.api import client as sdk_client
@@ -110,44 +111,46 @@ class TestWatchedSubprocess:
         rc = proc.wait()
 
         assert rc == 0
-        assert captured_logs == [
-            {
-                "chan": "stdout",
-                "event": "I'm a short message",
-                "level": "info",
-                "logger": "task",
-                "timestamp": "2024-11-07T12:34:56.078901Z",
-            },
-            {
-                "chan": "stderr",
-                "event": "stderr message",
-                "level": "error",
-                "logger": "task",
-                "timestamp": "2024-11-07T12:34:56.078901Z",
-            },
-            {
-                "chan": "stdout",
-                "event": "Message split across two writes",
-                "level": "info",
-                "logger": "task",
-                "timestamp": "2024-11-07T12:34:56.078901Z",
-            },
-            {
-                "event": "An error message",
-                "level": "error",
-                "logger": "airflow.foobar",
-                "timestamp": instant.replace(tzinfo=None),
-            },
-            {
-                "category": "UserWarning",
-                "event": "Warning should be captured too",
-                "filename": __file__,
-                "level": "warning",
-                "lineno": line,
-                "logger": "py.warnings",
-                "timestamp": instant.replace(tzinfo=None),
-            },
-        ]
+        assert captured_logs == unordered(
+            [
+                {
+                    "chan": "stdout",
+                    "event": "I'm a short message",
+                    "level": "info",
+                    "logger": "task",
+                    "timestamp": "2024-11-07T12:34:56.078901Z",
+                },
+                {
+                    "chan": "stderr",
+                    "event": "stderr message",
+                    "level": "error",
+                    "logger": "task",
+                    "timestamp": "2024-11-07T12:34:56.078901Z",
+                },
+                {
+                    "chan": "stdout",
+                    "event": "Message split across two writes",
+                    "level": "info",
+                    "logger": "task",
+                    "timestamp": "2024-11-07T12:34:56.078901Z",
+                },
+                {
+                    "event": "An error message",
+                    "level": "error",
+                    "logger": "airflow.foobar",
+                    "timestamp": instant.replace(tzinfo=None),
+                },
+                {
+                    "category": "UserWarning",
+                    "event": "Warning should be captured too",
+                    "filename": __file__,
+                    "level": "warning",
+                    "lineno": line,
+                    "logger": "py.warnings",
+                    "timestamp": instant.replace(tzinfo=None),
+                },
+            ]
+        )
 
     def test_subprocess_sigkilled(self):
         main_pid = os.getpid()

